libs/s25main/desktops/dskGameInterface.cpp

```diff
@@ -203,6 +183,47 @@ void dskGameInterface::StartScrolling(const Position& mousePos)
     WINDOWMANAGER.SetCursor(Cursor::Scroll);
 }
 
+void dskGameInterface::ShowPersistentWindowsAfterSwitch()
+{
+    auto& windows = SETTINGS.windows.persistentSettings;
+
+    if(windows[CGI_CHAT].isOpen)
+        WINDOWMANAGER.ShowAfterSwitch(std::make_unique<iwChat>(this));
+    if(windows[CGI_POSTOFFICE].isOpen)
+        WINDOWMANAGER.ShowAfterSwitch(std::make_unique<iwPostWindow>(gwv, GetPostBox()));
+    if(windows[CGI_DISTRIBUTION].isOpen)
+        WINDOWMANAGER.ShowAfterSwitch(std::make_unique<iwDistribution>(gwv.GetViewer(), GAMECLIENT));
+    if(windows[CGI_BUILDORDER].isOpen && gwv.GetWorld().GetGGS().isEnabled(AddonId::CUSTOM_BUILD_SEQUENCE))
+        WINDOWMANAGER.ShowAfterSwitch(std::make_unique<iwBuildOrder>(gwv.GetViewer()));
+    if(windows[CGI_TRANSPORT].isOpen)
+        WINDOWMANAGER.ShowAfterSwitch(std::make_unique<iwTransport>(gwv.GetViewer(), GAMECLIENT));
+    if(windows[CGI_MILITARY].isOpen)
+        WINDOWMANAGER.ShowAfterSwitch(std::make_unique<iwMilitary>(gwv.GetViewer(), GAMECLIENT));
+    if(windows[CGI_TOOLS].isOpen)
+        WINDOWMANAGER.ShowAfterSwitch(std::make_unique<iwTools>(gwv.GetViewer(), GAMECLIENT));
+    if(windows[CGI_INVENTORY].isOpen)
+        WINDOWMANAGER.ShowAfterSwitch(std::make_unique<iwInventory>(gwv.GetViewer().GetPlayer()));
+    if(windows[CGI_MINIMAP].isOpen)
+        WINDOWMANAGER.ShowAfterSwitch(std::make_unique<iwMinimap>(minimap, gwv));
+    if(windows[CGI_BUILDINGS].isOpen)
+        WINDOWMANAGER.ShowAfterSwitch(std::make_unique<iwBuildings>(gwv, GAMECLIENT));
+    if(windows[CGI_BUILDINGSPRODUCTIVITY].isOpen)
+        WINDOWMANAGER.ShowAfterSwitch(std::make_unique<iwBuildingProductivities>(gwv.GetViewer().GetPlayer()));
+    if(windows[CGI_MUSICPLAYER].isOpen)
+        WINDOWMANAGER.ShowAfterSwitch(std::make_unique<iwMusicPlayer>());
+    if(windows[CGI_STATISTICS].isOpen)
+        WINDOWMANAGER.ShowAfterSwitch(std::make_unique<iwStatistics>(gwv.GetViewer()));
+    if(windows[CGI_ECONOMICPROGRESS].isOpen && gwv.GetWorld().getEconHandler())
+        WINDOWMANAGER.ShowAfterSwitch(std::make_unique<iwEconomicProgress>(gwv.GetViewer()));
+    if(windows[CGI_DIPLOMACY].isOpen)
+        WINDOWMANAGER.ShowAfterSwitch(std::make_unique<iwDiplomacy>(gwv.GetViewer(), GAMECLIENT));
+    if(windows[CGI_SHIP].isOpen)
+        WINDOWMANAGER.ShowAfterSwitch(
+          std::make_unique<iwShip>(gwv, GAMECLIENT, gwv.GetViewer().GetPlayer().GetShipByID(0)));
```

Comment:
does this really work? what if he opens a game where no ship is there?

Reply:
It seems weird, I agree. But it works and is in fact the same way the iwShip window is constructed when opened from iwMainMenu.